### PR TITLE
feat: implement initial setup of amplitude SDK

### PR DESCRIPTION
**Pull request by lynnshaoyu**
First PR for setting up amplitude SDK for use throughout web interface. 

Upcoming PR will implement Trace.tsx and TraceEvent.tsx, which will be react components that help log events and relevant context metadata, inspired by judo's framework implemented in mobile here: https://github.com/Uniswap/mobile/pull/362 

API Key committed within PR in env.production is for our amplitude TEST PROJECT. All events must initially be sent to our Test Project only. In dev environment (developer local env and vercel), nothing should be sent. 

Note: we have not removed Google Analytics (yet) and it will continue to exist in its separate folder, to be deleted at the end of this amplitude instrumentation work.

### CODE REVIEW

File: src/components/AmplitudeAnalytics/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  if (process.env.NODE_ENV === 'development') return
+
+  const API_KEY = process.env.REACT_APP_AMPLITUDE_KEY
+  if (typeof API_KEY === 'undefined') {
+    throw new Error(`REACT_APP_AMPLITUDE_KEY must be a defined environment variable`)
+  }
+
+  init(
+    API_KEY,
+    undefined, // User ID should be undefined to let Amplitude default to Device ID
```
Comment (vm) [inline src/components/AmplitudeAnalytics/index.ts:21]: perhaps name a variable called `const userId = undefined` to make this clearer

Comment (lynnshaoyu) [reply src/components/AmplitudeAnalytics/index.ts]: Added an inline comment for clarity, for both userId and options params. Thanks for pointing out.

File: src/components/AmplitudeAnalytics/index.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      identifyObj.preInsert(propertyName, propertyValue)
+      break
+    case UserPropertyOperations.ARRAY_POSTINSERT:
+      identifyObj.postInsert(propertyName, propertyValue)
+      break
+    case UserPropertyOperations.ARRAY_REMOVE:
+      identifyObj.remove(propertyName, propertyValue)
+      break
+  }
+  identify(identifyObj)
```
[the remark this answers is not in the document]
Comment (lynnshaoyu) [reply src/components/AmplitudeAnalytics/index.ts:88]: Same as above. Updated the function for handing dev environment.

File: src/components/AmplitudeAnalytics/index.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+/**
+ * Singleton that exposes methods to modify the User Model's properties in
+ * Amplitude that represents the current session's user.
+ *
+ * See https://help.amplitude.com/hc/en-us/articles/115002380567-User-properties-and-event-properties
+ * for details.
+ */
+export class UserModel {
```
[the remark this answers is not in the document]
Comment (lynnshaoyu) [reply src/components/AmplitudeAnalytics/index.ts]: yeah sounds good that's simpler, changed.

File: yarn.lock [hunk 4]
```diff
@@ -17705,6 +17733,11 @@ tslib@^2, tslib@^2.0.0, tslib@^2.0.3, tslib@^2.1.0, tslib@~2.3.0:
   resolved "https://registry.yarnpkg.com/tslib/-/tslib-2.3.1.tgz#e8a335add5ceae51aa261d32a490158ef042ef01"
   integrity sha512-77EbyPPpMz+FRFRuAFlWMtmgUWGe9UOG2Z25NqCwiIjRhOf5iKGuzSe5P2w1laq+FkRy4p+PCuVkJSGkzTEKVw==
 
+tslib@^2.3.1:
```
[the remark this answers is not in the document]
Comment (lynnshaoyu) [reply yarn.lock]: ah yes, thanks done.

File: src/components/AmplitudeAnalytics/index.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * Class that exposes methods to modify the User Model's properties in
+ * Amplitude that represents the current session's user.
+ *
+ * See https://help.amplitude.com/hc/en-us/articles/115002380567-User-properties-and-event-properties
+ * for details.
+ */
+class UserModel {
+  private _isDevEnvironemnt = true
```
Comment (cmcewen) [inline src/components/AmplitudeAnalytics/index.ts:52]: typo - `isDevEnvironment`

File: src/components/AmplitudeAnalytics/index.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      console.log('amplitude user model update with operation remove:', `${propertyName}: ${propertyValue}`)
+      return
+    }
+    const identifyObj = new Identify()
+    identifyObj.remove(propertyName, propertyValue)
+    identify(identifyObj)
+  }
+}
+
+export const userModel = new UserModel()
```
Comment (cmcewen) [inline src/components/AmplitudeAnalytics/index.ts:109]: my gut says this should be a capital case so how about changing the class to `_UserModel` and then you can `export const UserModel`. also i'm not tied to the name user model if you can think of something better.

**Review by cmcewen — APPROVED** [review]
overall looks good to me - would love a quick glance from @zzmp or @JFrankfurt too

**Comment by zzmp** [comment]
Missed the review window here, but made some small changes in https://github.com/Uniswap/interface/pull/4070.

**Pull request merged**